builder/builder.go

```diff
@@ -294,26 +357,24 @@ func (b *Builder) AddBundles(bundles []string, force bool, allbundles bool, git
 
 	// Check if mix bundles dir exists
 	if _, err := os.Stat(bundledir); os.IsNotExist(err) {
-		helpers.PrintError(errors.New("Mix bundles directory does not exist. Run mixer init-mix"))
-		os.Exit(1)
+		return bundleAddCount, errors.New("Mix bundles directory does not exist. Run mixer init-mix")
```

Comment:
Optional: unless there is a reason, I tend to return the zero values when returning errors. This signals to future code changes that they shouldn't care about keeping the count for this particular code path.

Reply:
Yeah, that's the convention I tried to follow elsewhere.  For this, though, the call affects the file system. If it thinks it has successfully added any bundles, that means those files actually now exist in the `mix-bundles` directory. As is, the user gets notified that bundles were in fact added, even if there was a failure along the way.

A cleaner implementation (for the user) would be to keep track of which bundles were actually added (as opposed to skipped for already being there, etc.) and attempt to rewind/clean-up in the event of an error. All of this is going away with the local/upstream bundle rewrite I'm doing this week, though, so I don't plan to implement that kind of bookkeeping for the time being.

Reply:
Optional: since there is a reason for those partial values even when there is an error, document it on the function comment something along the lines: "when there is error, the int value will contain the bundles added so far".

"All of this is going away with the local/upstream bundle rewrite I'm doing this week, though, so I don't plan to implement that kind of bookkeeping for the time being." -> OK!

Reply:
Fixed in forthcoming push.